### ISSUE TITLE
Enable Typechecking When Imported

### ISSUE THREAD

**Issue by mileswatson** [polygon-io/client-python#379]
**Is your feature request related to a problem? Please describe.**
When type checking a project that uses this library with MyPy, type checks are skipped for the import:

```
error: Skipping analyzing "polygon": module is installed, but missing library stubs or py.typed marker  [import]
note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
```

**Describe the solution you'd like**
Include an empty `py.typed` file in the root of the module. See: https://peps.python.org/pep-0561/#packaging-type-information 

**Describe alternatives you've considered**

Potential workarounds include disabling module typing entirely using `import polygon # type: ignore`, or using a script that automatically creates the py.typed file in the import before typechecking.

```py
from typing import Any
from mypy import api
import polygon


def enable_typing(lib: Any) -> None:  # type: ignore
    open(lib.__path__[0] + "/py.typed", 'a').close()


enable_typing(polygon)


result = api.run(["."])
if result[0]:
    print('\nType checking report:\n')
    print(result[0])

if result[1]:
    print('\nError report:\n')
    print(result[1])

print('\nExit status:', result[2])
quit(result[2])
```

Workarounds are fairly easy, but so is the solution so I figured I'd post this here.

**Additional context**
N/A

**Comment by justinpolygon** [comment]
Hey @mileswatson, we don't seem to implement py.typed on any of the python repos besides some examples code. I'm just wondering how much headache this is causing on your end, or maybe the larger community impact, you have any thoughts? I'm leaning towards closing saying we don't support it but wanted to check first.

**Comment by mileswatson** [comment]
Hi @justinpolygon! There's not too much *need* for this - the workarounds are good enough. It's nice because it allows typechecking and (arguably more importantly) editor code completion, and the actual change isn't too much work. 

However, I definitely understand if you don't want to commit to supporting typing - there are a few quirks with the current library I can see people complaining about that aren't really a problem otherwise (e.g. the `raw` optional param polluting all request return types with `| HttpResponse`, or having every response field be `x | None | Any` and so having to repeatedly assert `is not None`).

Basically just wanted to check this was a conscious decision and not something that was simply forgotten :)

**Comment by justinpolygon** [comment]
Great, thanks @mileswatson. I agree regarding having to use assert or is isinstance is a pain. We're thinking about improvements and I'll track this internally. I'm going to close the ticket for now but thanks for the feedback!

**Comment by ylhan** [comment]
@justinpolygon sorry to dig this up but +1 to adding type checking. We run `mypy` as a part of our CI/CD too.
